packages/frontend/src/hooks/validation/useValidation.ts

```diff
@@ -34,14 +39,23 @@ export const useValidationMutation = (projectUuid: string) => {
     const { showError } = useErrorLogs();
     const { showToastSuccess } = useToaster();
 
-    return useMutation<ValidationResponse[], ApiError>({
+    return useMutation<ApiJobScheduledResponse['results'], ApiError>({
         mutationKey: ['validation', projectUuid],
         mutationFn: () => updateValidation(projectUuid),
-        onSuccess: () => {
-            queryClient.invalidateQueries({ queryKey: ['validation'] });
-            showToastSuccess({
-                title: 'Success! Validation complete.',
-            });
+        onSuccess: (data) => {
+            // Wait until validation is complete
+            pollJobStatus(data.jobId)
+                .then(() => {
+                    // Invalidate validation to get latest results
+                    queryClient.invalidateQueries({ queryKey: ['validation'] });
+                    showToastSuccess({ title: 'Validation completed' });
+                })
+                .catch((error: Error) => {
+                    showError({
+                        title: 'Unable to update validation',
+                        body: error.message,
+                    });
```

Comment:
wonder if we need this `catch`, shouldn't it be caught by the `onError` below? 🤔

Reply:
Mmm, I think they are different , they are happening in different functions, onError will be happening on the `updateValidation` request, while this method is happening on the pollJobStatus (inside onSuccess)

Reply:
ignore! all good 🤦